### PR TITLE
Add some more CodeDom tests

### PR DESCRIPTION
**Pull request by hughbe**
Had these lying around for a while so finally got round to rebasing and sending them in

### CODE REVIEW

File: src/System.CodeDom/src/System/CodeDom/Compiler/CompilerInfo.cs [hunk 1]
```diff
@@ -84,14 +84,6 @@ public CodeDomProvider CreateProvider(IDictionary<string, string> providerOption
 
         public CompilerParameters CreateDefaultCompilerParameters() => CloneCompilerParameters();
 
-        internal CompilerInfo(CompilerParameters compilerParams, string codeDomProviderTypeName, string[] compilerLanguages, string[] compilerExtensions)
```
Comment (hughbe) [inline src/System.CodeDom/src/System/CodeDom/Compiler/CompilerInfo.cs]: also unused

File: src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs [hunk 2]
```diff
@@ -192,7 +189,7 @@ private int Indent
 
         private string NullToken => "null";
 
-        private CodeGeneratorOptions Options => _options;
+        private CodeGeneratorOptions Options { get; set; }
```
Comment (stephentoub) [inline src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: Why have a property at all rather than just having the field?

**Comment by stephentoub** [comment]
```
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CodeGeneratorTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CompilerInfoTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CodeGeneratorOptionsTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CompilerParametersTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CompilerErrorTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CompilerResultsTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
CSC : error CS2001: Source file 'D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System\CodeDom\Compiler\CodeCompilerTests.cs' could not be found. [D:\j\workspace\windows-TGrou---74aa877a\src\System.CodeDom\tests\System.CodeDom.Tests.csproj]
```

After fixing this, please run the unit tests before and after the change to verify that the exact same number are executed, such that no files were accidentally omitted.

**Comment by hughbe** [comment]
Before: 2139 tests ran
After: 2295 tests ran

File: src/System.CodeDom/tests/System/CodeDom/CodeNamespaceTests.cs [hunk 3]
```diff
@@ -3,13 +3,11 @@
...
 
 using System.Linq;
-using System.Reflection;
-using System.Runtime.Serialization;
 using Xunit;
 
 namespace System.CodeDom.Tests
 {
-	public class CodeNamespaceTests : CodeObjectTestBase<CodeNamespace>
+    public class CodeNamespaceTests : CodeObjectTestBase<CodeNamespace>
```
Comment (stephentoub) [inline src/System.CodeDom/tests/System/CodeDom/CodeNamespaceTests.cs]: Nit: indent looks off

File: src/System.CodeDom/tests/System/CodeDom/Compiler/CodeCompilerTests.cs [hunk 4]
```diff
@@ -0,0 +1,669 @@
...
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_NullFileNameInFileNames_ThrowsArgumentNullException()
+        {
+            ICodeCompiler compiler = new Compiler();
+            AssertExtensions.Throws<ArgumentNullException>("path", () => compiler.CompileAssemblyFromFileBatch(new CompilerParameters(), new string[] { null }));
+        }
+
+        [Fact]
+        public void CompileAssemblyFromFileBatch_EmptyFileNameInFileNames_ThrowsArgumentException()
```
Comment (stephentoub) [inline src/System.CodeDom/tests/System/CodeDom/Compiler/CodeCompilerTests.cs]: I expect this is going to fail when run on netfx... probably a few of the other tests as well.

Comment (hughbe) [reply src/System.CodeDom/tests/System/CodeDom/Compiler/CodeCompilerTests.cs]: Fixed

File: src/System.CodeDom/tests/System/CodeDom/CodeNamespaceTests.cs (hunk 3 above)
Comment (hughbe) [reply src/System.CodeDom/tests/System/CodeDom/CodeNamespaceTests.cs]: Fixed

**Comment by stephentoub** [comment]
Needs to be rebased. Thanks.

**Comment by stephentoub** [comment]
@dotnet-bot test this please ("ERROR: An error occurred when trying to initialize the tools." on all Linux legs)

**Pull request merged**